core/src/main/java/nu/yona/server/goals/service/GoalServiceException.java

```diff
@@ -25,9 +22,9 @@ public static GoalServiceException goalNotFoundById(UUID userID, UUID goalID)
 		return new GoalServiceException("error.goal.not.found.id", userID, goalID);
 	}
 
-	public static GoalServiceException cannotAddSecondGoalOnActivityCategory(String activityCategoryName)
+	public static GoalServiceException cannotAddSecondGoalOnActivityCategory(UUID activityCategoryID)
 	{
-		return new GoalServiceException("error.goal.cannot.add.second.on.activity.category", activityCategoryName);
+		return new GoalServiceException("error.goal.cannot.add.second.on.activity.category", activityCategoryID);
```

Comment:
The message was still holding good. However, it's pointless to send an activity category ID back to the user, so I've changed it back to the activity category name.